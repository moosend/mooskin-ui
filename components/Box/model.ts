import { IMooskinContext, IStyledTheme, IThemeBackgroundColors, IThemeBorderColors, IThemeFontColors } from '../Styled/model';
// import { IMooskinContext } from '../Styled/model';

export type IntensityType = 'xs' | 'sm' | 'base' | 'md' | 'lg' | 'xl' | '2xl';
export type BoxShadowIntensityType = IntensityType | 'inner' | 'outline';
export type NestedThemeType = `${keyof IStyledTheme}.${keyof IThemeFontColors | keyof IThemeBorderColors | keyof IThemeBackgroundColors}`;
// export type NestedThemeType = [string, string];

export const boxComponentProps = [
	'id',
	'className',
	'style',
	'm',
	'mt',
	'mr',
	'mb',
	'ml',
	'mx',
	'my',
	'boxShadow',
	'color',
	'round',
	'p',
	'pt',
	'pr',
	'pb',
	'pl',
	'px',
	'py',
	'opacity',
	'fontFamily',
	'fontSize',
	'fontWeight',
	'lineHeight',
	'textAlign',
	'fontStyle',
	'textTransform',
	'textDecoration',
	'w',
	'h',
	'minW',
	'maxW',
	'minH',
	'maxH',
	'd',
	'overflow',
	'overflowX',
	'overflowY',
	'align',
	'alignContent',
	'alignSelf',
	'justifyItems',
	'justify',
	'justifySelf',
	'wrap',
	'direction',
	'flex',
	'flexGrow',
	'flexShrink',
	'flexBasis',
	'order',
	'bg',
	'bgImage',
	'bgColor',
	'bgSize',
	'bgPosition',
	'bgRepeat',
	'border',
	'borderWidth',
	'borderStyle',
	'borderColor',
	'borderTop',
	'borderTopWidth',
	'borderTopStyle',
	'borderTopColor',
	'borderRight',
	'borderRightWidth',
	'borderRightStyle',
	'borderRightColor',
	'borderBottom',
	'borderBottomWidth',
	'borderBottomStyle',
	'borderBottomColor',
	'borderLeft',
	'borderLeftWidth',
	'borderLeftStyle',
	'borderLeftColor',
	'borderRadius',
	'borderTopLeftRadius',
	'borderTopRightRadius',
	'borderBottomRightRadius',
	'borderBottomLeftRadius',
	'position',
	'zIndex',
	'top',
	'right',
	'bottom',
	'left',
	'animation',
	'visibility',
	'cursor',
	'boxAs',
	'children',
	'theme',
];

export type IBoxComponentProps = IBaseBoxComponentProps & React.HTMLAttributes<HTMLElement>;
export type IBoxImageComponentProps = IBaseBoxComponentProps & React.ImgHTMLAttributes<HTMLImageElement>;
export type IBoxLabelComponentProps = IBaseBoxComponentProps & React.LabelHTMLAttributes<HTMLLabelElement>;
export type IButtonBoxComponentProps = IBoxComponentProps & React.ButtonHTMLAttributes<HTMLButtonElement>;
export type IInputBoxComponentProps = IBoxComponentProps & React.InputHTMLAttributes<HTMLInputElement>;
export type IAnchorBoxComponentProps = IBoxComponentProps & React.AnchorHTMLAttributes<HTMLAnchorElement>;
export type ITextAreaBoxComponentProps = IBoxComponentProps & React.TextareaHTMLAttributes<HTMLTextAreaElement>;

export interface IBaseBoxComponentProps extends IMooskinContext {
	ref?: any;

	setRef?: (ref: React.RefObject<HTMLElement>) => void;
	/** render Box as a different html element */
	boxAs?: any;

	/** margin */
	m?: React.CSSProperties['margin'] | Array<React.CSSProperties['margin']>;

	/** margin top */
	mt?: React.CSSProperties['marginTop'] | Array<React.CSSProperties['marginTop']>;

	/** margin right */
	mr?: React.CSSProperties['marginRight'] | Array<React.CSSProperties['marginRight']>;

	/** margin bottom */
	mb?: React.CSSProperties['marginBottom'] | Array<React.CSSProperties['marginBottom']>;

	/** margin left */
	ml?: React.CSSProperties['marginLeft'] | Array<React.CSSProperties['marginLeft']>;

	/** margin left & margin right */
	mx?: number | string | Array<number | string>;

	/** margin top & margin-bottom */
	my?: number | string | Array<number | string>;

	/** padding */
	p?: React.CSSProperties['padding'] | Array<React.CSSProperties['padding']>;

	/** padding top */
	pt?: React.CSSProperties['paddingTop'] | Array<React.CSSProperties['paddingTop']>;

	/** padding right */
	pr?: React.CSSProperties['paddingRight'] | Array<React.CSSProperties['paddingRight']>;

	/** padding bottom */
	pb?: React.CSSProperties['paddingBottom'] | Array<React.CSSProperties['paddingBottom']>;

	/** padding left */
	pl?: React.CSSProperties['paddingLeft'] | Array<React.CSSProperties['paddingLeft']>;

	/** padding left & padding right */
	px?: number | string | Array<number | string>;

	/** padding top & padding-bottom */
	py?: number | string | Array<number | string>;

	/** font color */
	fontColor?: React.CSSProperties['color'] | NestedThemeType | Array<React.CSSProperties['color']>;

	/** opacity */
	opacity?: React.CSSProperties['opacity'] | Array<React.CSSProperties['opacity']>;

	/** font family */
	fontFamily?: React.CSSProperties['fontFamily'] | Array<React.CSSProperties['fontFamily']>;

	/** font size */
	fontSize?: React.CSSProperties['fontSize'] | Array<React.CSSProperties['fontSize']>;

	/** font weight */
	fontWeight?: React.CSSProperties['fontWeight'] | Array<React.CSSProperties['fontWeight']>;

	/** line height */
	lineHeight?: React.CSSProperties['lineHeight'] | Array<React.CSSProperties['lineHeight']>;

	/** text align */
	textAlign?: React.CSSProperties['textAlign'] | Array<React.CSSProperties['textAlign']>;

	/** font style */
	fontStyle?: React.CSSProperties['fontStyle'] | Array<React.CSSProperties['fontStyle']>;

	/** text transform */
	textTransform?: React.CSSProperties['textTransform'] | Array<React.CSSProperties['textTransform']>;

	/** text decoration */
	textDecoration?: React.CSSProperties['textDecoration'] | Array<React.CSSProperties['textDecoration']>;

	/** width */
	w?: React.CSSProperties['width'] | Array<React.CSSProperties['width']>;

	/** height */
	h?: React.CSSProperties['height'] | Array<React.CSSProperties['height']>;

	/** max width */
	maxW?: React.CSSProperties['maxWidth'] | Array<React.CSSProperties['maxWidth']>;

	/** min width */
	minW?: React.CSSProperties['minWidth'] | Array<React.CSSProperties['minWidth']>;

	/** max height */
	maxH?: React.CSSProperties['maxHeight'] | Array<React.CSSProperties['maxHeight']>;

	/** min height */
	minH?: React.CSSProperties['minHeight'] | Array<React.CSSProperties['minHeight']>;

	/** display */
	d?: React.CSSProperties['display'] | Array<React.CSSProperties['display']>;

	/** overflow */
	overflow?: React.CSSProperties['overflow'] | Array<React.CSSProperties['overflow']>;

	/** overflow X */
	overflowX?: React.CSSProperties['overflowX'] | Array<React.CSSProperties['overflowX']>;

	/** overflow Y */
	overflowY?: React.CSSProperties['overflowY'] | Array<React.CSSProperties['overflowY']>;

	/** align items */
	align?: React.CSSProperties['alignItems'] | Array<React.CSSProperties['alignItems']>;

	/** align content */
	alignContent?: React.CSSProperties['alignContent'] | Array<React.CSSProperties['alignContent']>;

	/** align self */
	alignSelf?: React.CSSProperties['alignSelf'] | Array<React.CSSProperties['alignSelf']>;

	/** justify items */
	justifyItems?: React.CSSProperties['justifyItems'] | Array<React.CSSProperties['justifyItems']>;

	/** justify content */
	justify?: React.CSSProperties['justifyContent'] | Array<React.CSSProperties['justifyContent']>;

	/** justify self */
	justifySelf?: React.CSSProperties['justifySelf'] | Array<React.CSSProperties['justifySelf']>;

	/** flex wrap */
	flexWrap?: React.CSSProperties['flexWrap'] | Array<React.CSSProperties['flexWrap']>;

	/** flex direction */
	direction?: React.CSSProperties['flexDirection'] | Array<React.CSSProperties['flexDirection']>;

	/** flex */
	flex?: React.CSSProperties['flex'] | Array<React.CSSProperties['flex']>;

	/** flex grow */
	flexGrow?: React.CSSProperties['flexGrow'] | Array<React.CSSProperties['flexGrow']>;

	/** flex shrink */
	flexShrink?: React.CSSProperties['flexShrink'] | Array<React.CSSProperties['flexShrink']>;

	/** flex basis */
	flexBasis?: React.CSSProperties['flexBasis'] | Array<React.CSSProperties['flexBasis']>;

	/** order */
	order?: React.CSSProperties['order'] | Array<React.CSSProperties['order']>;

	/** background */
	bg?: React.CSSProperties['background'] | Array<React.CSSProperties['background']>;

	/** background image */
	bgImage?: React.CSSProperties['backgroundImage'] | Array<React.CSSProperties['backgroundImage']>;

	/** background color */
	bgColor?: React.CSSProperties['backgroundColor'] | NestedThemeType | Array<React.CSSProperties['backgroundColor']>;

	/** background size */
	bgSize?: React.CSSProperties['backgroundSize'] | Array<React.CSSProperties['backgroundSize']>;

	/** background position */
	bgPosition?: React.CSSProperties['backgroundPosition'] | Array<React.CSSProperties['backgroundPosition']>;

	/** background repeat */
	bgRepeat?: React.CSSProperties['backgroundRepeat'] | Array<React.CSSProperties['backgroundRepeat']>;

	/** border */
	border?: React.CSSProperties['border'] | Array<React.CSSProperties['border']>;

	/** border width */
	borderWidth?: React.CSSProperties['borderWidth'] | Array<React.CSSProperties['borderWidth']>;

	/** border style */
	borderStyle?: React.CSSProperties['borderStyle'] | Array<React.CSSProperties['borderStyle']>;

	/** border color */
	borderColor?: React.CSSProperties['borderColor'] | NestedThemeType | Array<React.CSSProperties['borderColor']>;

	/** border top */
	borderTop?: React.CSSProperties['borderTop'] | Array<React.CSSProperties['borderTop']>;

	/** border top width */
	borderTopWidth?: React.CSSProperties['borderTopWidth'] | Array<React.CSSProperties['borderTopWidth']>;

	/** border top style */
	borderTopStyle?: React.CSSProperties['borderTopStyle'] | Array<React.CSSProperties['borderTopStyle']>;

	/** border top color */
	borderTopColor?: React.CSSProperties['borderTopColor'] | Array<React.CSSProperties['borderTopColor']>;

	/** border right */
	borderRight?: React.CSSProperties['borderRight'] | Array<React.CSSProperties['borderRight']>;

	/** border right width */
	borderRightWidth?: React.CSSProperties['borderRightWidth'] | Array<React.CSSProperties['borderRightWidth']>;

	/** border right style */
	borderRightStyle?: React.CSSProperties['borderRightStyle'] | Array<React.CSSProperties['borderRightStyle']>;

	/** border right color */
	borderRightColor?: React.CSSProperties['borderRightColor'] | Array<React.CSSProperties['borderRightColor']>;

	/** border bottom */
	borderBottom?: React.CSSProperties['borderBottom'] | Array<React.CSSProperties['borderBottom']>;

	/** border bottom width */
	borderBottomWidth?: React.CSSProperties['borderBottomWidth'] | Array<React.CSSProperties['borderBottomWidth']>;

	/** border bottom style */
	borderBottomStyle?: React.CSSProperties['borderBottomStyle'] | Array<React.CSSProperties['borderBottomStyle']>;

	/** border bottom color */
	borderBottomColor?: React.CSSProperties['borderBottomColor'] | Array<React.CSSProperties['borderBottomColor']>;

	/** border left */
	borderLeft?: React.CSSProperties['borderLeft'] | Array<React.CSSProperties['borderLeft']>;

	/** border left width */
	borderLeftWidth?: React.CSSProperties['borderLeftWidth'] | Array<React.CSSProperties['borderLeftWidth']>;

	/** border left style */
	borderLeftStyle?: React.CSSProperties['borderLeftStyle'] | Array<React.CSSProperties['borderLeftStyle']>;

	/** border left color */
	borderLeftColor?: React.CSSProperties['borderLeftColor'] | Array<React.CSSProperties['borderLeftColor']>;

	/** border radius */
	borderRadius?: React.CSSProperties['borderRadius'] | Array<React.CSSProperties['borderRadius']>;

	/** border radius top left */
	borderTopLeftRadius?: React.CSSProperties['borderTopLeftRadius'] | Array<React.CSSProperties['borderTopLeftRadius']>;

	/** border radius top right */
	borderTopRightRadius?: React.CSSProperties['borderTopRightRadius'] | Array<React.CSSProperties['borderTopRightRadius']>;

	/** border radius bottom right */
	borderBottomRightRadius?: React.CSSProperties['borderBottomRightRadius'] | Array<React.CSSProperties['borderBottomRightRadius']>;

	/** border radius bottom left */
	borderBottomLeftRadius?: React.CSSProperties['borderBottomLeftRadius'] | Array<React.CSSProperties['borderBottomLeftRadius']>;

	/** position */
	position?: React.CSSProperties['position'] | Array<React.CSSProperties['position']>;

	/** z index */
	zIndex?: React.CSSProperties['zIndex'] | Array<React.CSSProperties['zIndex']>;

	/** top */
	top?: React.CSSProperties['top'] | Array<React.CSSProperties['top']>;

	/** right */
	right?: React.CSSProperties['right'] | Array<React.CSSProperties['right']>;

	/** bottom */
	bottom?: React.CSSProperties['bottom'] | Array<React.CSSProperties['bottom']>;

	/** left */
	left?: React.CSSProperties['left'] | Array<React.CSSProperties['left']>;

	/** box shadow */
	// boxShadow?: React.CSSProperties['boxShadow'] | BoxShadowIntensityType | Array<React.CSSProperties['boxShadow']>;
	boxShadow?: React.CSSProperties['boxShadow'] | BoxShadowIntensityType;

	/** animation */
	animation?: React.CSSProperties['animation'] | Array<React.CSSProperties['animation']>;

	/** visibility */
	visibility?: React.CSSProperties['visibility'] | Array<React.CSSProperties['visibility']>;

	/** cursor */
	cursor?: React.CSSProperties['cursor'] | Array<React.CSSProperties['cursor']>;

	/** rounds the element */
	// round?: IntensityType | IntensityType[];
	round?: IntensityType;

	/** white space */
	whiteSpace?: React.CSSProperties['whiteSpace'] | Array<React.CSSProperties['whiteSpace']>;

	/** word break */
	wordBreak?: React.CSSProperties['wordBreak'] | Array<React.CSSProperties['wordBreak']>;

	/** grid template columns */
	gridTemplateColumns?: React.CSSProperties['gridTemplateColumns'] | Array<React.CSSProperties['gridTemplateColumns']>;

	/** hover */
	_hover?: string;

	/** all */
	all?: React.CSSProperties['all'] | Array<React.CSSProperties['all']>;

	/** transition */
	transition?: React.CSSProperties['transition'] | Array<React.CSSProperties['transition']>;

	/** transition */
	transform?: React.CSSProperties['transform'] | Array<React.CSSProperties['transform']>;

	/** text overflow */
	textOverflow?: React.CSSProperties['textOverflow'] | Array<React.CSSProperties['textOverflow']>;

	/** outline */
	outline?: React.CSSProperties['outline'] | Array<React.CSSProperties['outline']>;

	/** outline width */
	outlineWidth?: React.CSSProperties['outlineWidth'] | Array<React.CSSProperties['outlineWidth']>;

	/** outline style */
	outlineStyle?: React.CSSProperties['outlineStyle'] | Array<React.CSSProperties['outlineStyle']>;

	/** outline color */
	outlineColor?: React.CSSProperties['outlineColor'] | NestedThemeType | Array<React.CSSProperties['outlineColor']>;
}
