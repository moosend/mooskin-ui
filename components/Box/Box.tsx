import * as React from 'react';

// Models
import { IBoxComponentProps, ScreenType } from './model';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Styled Components
import { StyledBox } from './styles';
import { Screens } from '../_utils/globals/screens';

/**
 * Box
 */
export const Box: React.FC<IBoxComponentProps> = withMooskinContext(
	React.memo((props) => {
		const [shouldRender, setShouldRender] = React.useState(true);

		React.useEffect(() => {
			if (props.noRender) {
				setShouldRender(checkShouldRender(props.noRender));

				const onResize = () => {
					if (props.noRender) setShouldRender(checkShouldRender(props.noRender));
				};

				window.addEventListener('resize', onResize);

				return () => {
					window.removeEventListener('resize', onResize);
				};
			}
		}, []);

		if (shouldRender) {
			return <StyledBox ref={(ref: HTMLElement) => props.setRef && props.setRef(ref)} {...props} as={props.boxAs} />;
		}

		return null;
	})
);

const checkShouldRender = (noRender: Array<ScreenType>) => {
	const width = window.innerWidth;

	const found = noRender.find((screen) => {
		const limits = Screens[screen];
		return limits.max >= width && limits.min <= width;
	});

	if (found) return false;

	return true;
};

Box.defaultProps = {
	className: '',
	style: {},
	transition: '0.2s all'
};

Box.displayName = 'Box';
