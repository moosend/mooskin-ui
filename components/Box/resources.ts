interface ISupportedStyleProps {
	[key: string]: {
		property: string;
		processedValue?: boolean;
		usePallete?: boolean;
	};
}

export const SupportedStyleProps: ISupportedStyleProps = {
	m: {
		processedValue: true,
		property: 'margin',
	},
	mt: {
		processedValue: true,
		property: 'margin-top',
	},
	mr: {
		processedValue: true,
		property: 'margin-right',
	},
	mb: {
		processedValue: true,
		property: 'margin-bottom',
	},
	ml: {
		processedValue: true,
		property: 'margin-left',
	},
	p: {
		processedValue: true,
		property: 'padding',
	},
	pt: {
		processedValue: true,
		property: 'padding-top',
	},
	pr: {
		processedValue: true,
		property: 'padding-right',
	},
	pb: {
		processedValue: true,
		property: 'padding-bottom',
	},
	pl: {
		processedValue: true,
		property: 'padding-left',
	},
	fontColor: {
		property: 'color',
		usePallete: true,
	},
	fontFamily: {
		property: 'font-family',
	},
	fontSize: {
		property: 'font-size',
	},
	fontWeight: {
		property: 'font-weight',
	},
	lineHeight: {
		property: 'line-height',
	},
	textAlign: {
		property: 'text-align',
	},
	fontStyle: {
		property: 'font-style',
	},
	textTransform: {
		property: 'text-transform',
	},
	textDecoration: {
		property: 'text-decoration',
	},
	whiteSpace: {
		property: 'white-space',
	},
	w: {
		processedValue: true,
		property: 'width',
	},
	h: {
		processedValue: true,
		property: 'height',
	},
	minW: {
		processedValue: true,
		property: 'min-width',
	},
	maxW: {
		processedValue: true,
		property: 'max-width',
	},
	minH: {
		processedValue: true,
		property: 'min-height',
	},
	maxH: {
		processedValue: true,
		property: 'max-height',
	},
	d: {
		property: 'display',
	},
	overflow: {
		property: 'overflow',
	},
	overflowY: {
		property: 'overflow-y',
	},
	overflowX: {
		property: 'overflow-x',
	},
	align: {
		property: 'align-items',
	},
	alignContent: {
		property: 'align-content',
	},
	alignSelf: {
		property: 'align-self',
	},
	justify: {
		property: 'justify-content',
	},
	justifyItems: {
		property: 'justify-items',
	},
	justifySelf: {
		property: 'justify-self',
	},
	flexWrap: {
		property: 'flex-wrap',
	},
	direction: {
		property: 'flex-direction',
	},
	flex: {
		property: 'flex',
	},
	flexGrow: {
		property: 'flex-grow',
	},
	flexShrink: {
		property: 'flex-shrink',
	},
	flexBasis: {
		processedValue: true,
		property: 'flex-basis',
	},
	order: {
		property: 'order',
	},
	bg: {
		property: 'background',
	},
	bgImage: {
		property: 'background-image',
	},
	bgColor: {
		property: 'background-color',
		usePallete: true,
	},
	bgSize: {
		property: 'background-size',
	},
	bgPosition: {
		property: 'background-position',
	},
	bgRepeat: {
		property: 'background-repeat',
	},
	border: {
		property: 'border',
	},
	borderStyle: {
		property: 'border-style',
	},
	borderColor: {
		property: 'border-color',
		usePallete: true,
	},
	borderWidth: {
		processedValue: true,
		property: 'border-width',
	},
	borderTop: {
		property: 'border-top',
	},
	borderTopStyle: {
		property: 'border-top-style',
	},
	borderTopColor: {
		property: 'border-top-color',
		usePallete: true,
	},
	borderTopWidth: {
		processedValue: true,
		property: 'border-top-width',
	},
	borderRight: {
		property: 'border-right',
	},
	borderRightStyle: {
		property: 'border-right-style',
	},
	borderRightColor: {
		property: 'border-right-color',
		usePallete: true,
	},
	borderRightWidth: {
		processedValue: true,
		property: 'border-right-width',
	},
	borderBottom: {
		property: 'border-bottom',
	},
	borderBottomStyle: {
		property: 'border-bottom-style',
	},
	borderBottomColor: {
		property: 'border-bottom-color',
		usePallete: true,
	},
	borderBottomWidth: {
		processedValue: true,
		property: 'border-bottom-width',
	},
	borderLeft: {
		property: 'border-left',
	},
	borderLeftStyle: {
		property: 'border-left-style',
	},
	borderLeftColor: {
		property: 'border-left-color',
		usePallete: true,
	},
	borderLeftWidth: {
		processedValue: true,
		property: 'border-left-width',
	},
	borderRadius: {
		processedValue: true,
		property: 'border-radius',
	},
	borderTopLeftRadius: {
		processedValue: true,
		property: 'border-top-left-radius',
	},
	borderTopRightRadius: {
		processedValue: true,
		property: 'border-top-right-radius',
	},
	borderBottomRightRadius: {
		processedValue: true,
		property: 'border-bottom-right-radius',
	},
	borderBottomLeftRadius: {
		processedValue: true,
		property: 'border-bottom-left-radius',
	},
	position: {
		property: 'position',
	},
	zIndex: {
		property: 'z-index',
	},
	top: {
		processedValue: true,
		property: 'top',
	},
	right: {
		processedValue: true,
		property: 'right',
	},
	bottom: {
		processedValue: true,
		property: 'bottom',
	},
	left: {
		processedValue: true,
		property: 'left',
	},
	boxShadow: {
		property: 'box-shadow',
	},
	animation: {
		property: 'animation',
	},
	visibility: {
		property: 'visibility',
	},
	opacity: {
		property: 'opacity',
	},
	cursor: {
		property: 'cursor',
	},
	all: {
		property: 'all',
	},
	transition: {
		property: 'transition',
	},
	textOverflow: {
		property: 'text-overflow',
	},
	gridTemplateColumns: {
		property: 'grid-template-columns',
	},
	wordBreak: {
		property: 'word-break',
	},
};
