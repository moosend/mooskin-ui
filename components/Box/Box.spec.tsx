import * as React from 'react';
import { Box } from './Box';

import { mount } from 'enzyme';

describe('Box', () => {
	test('renders Box correctly', () => {
		const tree = mount(
			<Box align="baseline" p={5} m={5}>
				Box
			</Box>
		);
		expect(tree).toMatchSnapshot();
	});
});
