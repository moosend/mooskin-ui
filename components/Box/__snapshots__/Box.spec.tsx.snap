// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders Box correctly 1`] = `
<Box
  align="baseline"
  className=""
  m={5}
  p={5}
  style={Object {}}
  transition="0.2s all"
>
  <Component
    align="baseline"
    className=""
    m={5}
    p={5}
    palette={
      Object {
        "backgroundColors": Object {
          "accountNavigation": "#FFFFFF",
          "altWhite": "#FFFFFF",
          "autosExpression": "#212121",
          "background": "#FAFAFA",
          "black": "#000000",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "client": "#E0E0E0",
          "createNew": "#FFFFFF",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "submenu": "#FFFFFF",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "accountNavigation": "#FFFFFF",
          "altWhite": "#FFFFFF",
          "autosExpression": "#212121",
          "background": "#FAFAFA",
          "black": "#000000",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "client": "#E0E0E0",
          "createNew": "#FFFFFF",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "submenu": "#FFFFFF",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "accountNavigation": "#FFFFFF",
          "altWhite": "#FFFFFF",
          "autosExpression": "#212121",
          "background": "#FAFAFA",
          "black": "#000000",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "client": "#E0E0E0",
          "createNew": "#212121",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#212121",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "submenu": "#FFFFFF",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    style={Object {}}
    transition="0.2s all"
  >
    <StyledBox
      align="baseline"
      className=""
      m={5}
      p={5}
      palette={
        Object {
          "backgroundColors": Object {
            "accountNavigation": "#FFFFFF",
            "altWhite": "#FFFFFF",
            "autosExpression": "#212121",
            "background": "#FAFAFA",
            "black": "#000000",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "client": "#E0E0E0",
            "createNew": "#FFFFFF",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "submenu": "#FFFFFF",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "accountNavigation": "#FFFFFF",
            "altWhite": "#FFFFFF",
            "autosExpression": "#212121",
            "background": "#FAFAFA",
            "black": "#000000",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "client": "#E0E0E0",
            "createNew": "#FFFFFF",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "submenu": "#FFFFFF",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "accountNavigation": "#FFFFFF",
            "altWhite": "#FFFFFF",
            "autosExpression": "#212121",
            "background": "#FAFAFA",
            "black": "#000000",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "client": "#E0E0E0",
            "createNew": "#212121",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#212121",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "submenu": "#FFFFFF",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
      style={Object {}}
      transition="0.2s all"
    >
      <div
        className="sc-bdnylx gKsrPu"
        style={Object {}}
      >
        Box
      </div>
    </StyledBox>
  </Component>
</Box>
`;
